File: doc-examples/nitrite-example-java/src/main/java/example/ProjectRepository.java
package example;

import io.micronaut.data.nitrite.annotation.NitriteRepository;
import io.micronaut.data.repository.CrudRepository;

// tag::projectRepository[]
@NitriteRepository
public interface ProjectRepository extends CrudRepository<Project, ProjectId> {
}
// end::projectRepository[]

File: doc-examples/nitrite-example-kotlin/src/main/kotlin/example/ProjectRepository.kt
package example

import io.micronaut.data.nitrite.annotation.NitriteRepository
import io.micronaut.data.repository.CrudRepository
import java.util.Optional

// tag::projectRepository[]
@NitriteRepository
interface ProjectRepository : CrudRepository<Project, ProjectId> {

    fun findByName(name: String): Optional<Project>
}
// end::projectRepository[]
